### DISCUSSION TITLE
Height or Width of the Ellipse as Proxy for the Pupil Diameter?

### DISCUSSION THREAD

**Discussion by Hedenir** [openPupil/Open-PupilEXT#19] [General]
Hi Zand, congratulations for your PupilEXT project.

I would like ask why you considered the hight as pupil diameter and not  the width.
There is a particular reason?

Thanks

**Comment by Hedenir** [comment]
Hi,
I saw that you have answeard this question in previus discussion:
"In pupillometry, the longer axis (major axis) is always used as the pupil diameter to minimise pupil foreshortening error."

This seems to be very logical and clear for me. Thanks a lot.
Do you have a reference for this?

**Comment by Hedenir** [comment]
Hi,
Two more questions:

1-Some times the programa don't stop after analise a folder with jpg files and continue in a infinity loop. Do you know why?
2-In off-line analise, is there a way to mesure the deameters direct from a video (.mp4)? Without need to extrat frames .jpg?

**Comment by BZandi** [comment]
In the preference window of PupilEXT, you can toggle the option of whether you want an infinite loop or not during the offline analysis (see the attached image below).

![PupilEXT_Setting](https://user-images.githubusercontent.com/62992112/153145919-d0172722-a4b8-46cc-b1a5-700b5a02d2f7.png)

Analyzing *mp4 files is not possible, but there are plenty of tools that you can use to extract the images from such video files. However, you need to ensure that the image files are appropriately labelled. Please read section "2.4 Demo dataset for offline analysis" in the readme file of PupilEXT (see [here](https://github.com/openPupil/Open-PupilEXT)) as there is a note on how to label the images properly when using externally acquired images. 

Another issue might be when adjusting the processing framerate (in offline mode) higher than your computer can handle. In such cases, the images will be saved in the computer's memory and processed step by step. Two numbers are displayed in the live view panel, one for the processing speed and one for the camera's fps count. These two values need to be approximately on the same level; otherwise, the images will be saved in the buffer. Consequently, if you press "stop", the live view will not stop immediately, as PupilEXT is processing the remaining images in the buffer. 

Regarding pupil foreshortening error, the following references might be interesting for you:

1) **Mapping and correcting the influence of gaze position on pupil size measurements**
[https://pubmed.ncbi.nlm.nih.gov/25953668/](https://pubmed.ncbi.nlm.nih.gov/25953668/)

2) **Gaze-angle dependency of pupil-size measurements in head-mounted eye tracking**
[https://link.springer.com/article/10.3758/s13428-021-01657-8](https://link.springer.com/article/10.3758/s13428-021-01657-8)

Why the major axis is the better choice can also be simulated. For example, you can use an eye model to rotate the eye while keeping the pupil diameter steady. You will see that the major axis of the fitted ellipse is less affected by the rotation of the eye.

Geoffrey Aguirre recently published a model which could be leveraged for such a simulation:

3) **A model of the entrance pupil of the human eye**
[https://www.nature.com/articles/s41598-019-45827-3](https://www.nature.com/articles/s41598-019-45827-3)
